File: Python/String-1/right2.py

#given a string, return a "rotated right 2" version where the last 2 chars are moved to the start
# the string length will be at least 2
def right2(str):
    s = ''

    if len(str) > 1:
        s = str[-2:] + str[:-2]

    return s

print(right2('Hello'))
print(right2('java'))
print(right2('Hi'))

File: Java/String-1/right2.java

public class string1 {
	
	// given a string, return a "rotated right 2" version where the last 2 chars are moved to the start
	// the string length will be at least 2
	public String right2(String str) {
		
		String s = "";
		
		if (str.length() > 1)
			s = str.substring(str.length()-2) + str.substring(0, str.length()-2);
		
		return s;
		
	}

	public static void main(String[] args) {
		
		string1 o = new string1();
		
		System.out.println(o.right2("Hello"));
		System.out.println(o.right2("java"));
		System.out.println(o.right2("Hi"));
	
	}

}
